src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -111,7 +111,7 @@ public function getTypes($class, $property, array $context = [])
         }
 
         if (
-            $context['enable_constructor_extraction'] ?? $this->enableConstructorExtraction &&
+            ($context['enable_constructor_extraction'] ?? $this->enableConstructorExtraction) &&
```

Comment:
are you sure your tests only pass if you add these brackets? 🤔

Reply:
without the brackets: 
1) Symfony\Component\PropertyInfo\Tests\Extractor\ReflectionExtractorTest::testExtractTypeConstructor with data set #0 ('Symfony\Component\PropertyInf...1Dummy', 'string', array(Symfony\Component\PropertyInfo\Type Object (...)))
Undefined variable: fromConstructor


if I knock up a quick script, returning true for $a, the bellows returns null / 1:

$a = $b = $c = 1;

if( $a ?? $b && $d = $c ) {
	var_dump($d);
}
if( ($a ?? $b) && $d = $c ) {
	var_dump($d);
}

returns null / 1